### PR TITLE
refactor: migrate to typescript, fixes #42

### LINKED ISSUE #42: Add types for Typescript
## Description
Add typescript types to this library.
Preferably in the form of a built in `.d.ts` file or otherwise as a separate `@types/typeform` package.

## Context
In #25, @MichaelSolati pushed a complete rewrite of the library.

Rather than changing the current codebase, a type definition of the current API would be more than adequate!
It would benefit Typeform's API ecosystem, current and potential developers to have a strict, public type definition of the API.

## Possible Solution
Breifly looking at `src`, it seems this library isn't automatically generated based of a spec (correct me if i'm wrong).
The README is well defined though. Assuming there's some sort of internal definition for the backend, it could be useful to manually or automatically generate the type surface.

## Current Solution

```ts
//@ts-ignore ... 😒
import { createClient } from '@typeform/api-client'
```

### PR DESCRIPTION
**Pull request by MichaelSolati**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,61 +52,39 @@
...
-    "babel-jest": "^22.4.3",
-    "babel-loader": "^7.1.4",
-    "eslint": "^5.16.0",
-    "eslint-config-standard": "^12.0.0",
-    "eslint-plugin-import": "^2.17.2",
-    "eslint-plugin-node": "^8.0.1",
-    "eslint-plugin-promise": "^4.1.1",
-    "eslint-plugin-standard": "^4.0.0",
-    "husky": "^2.1.0",
+    "husky": "^4.0.0-beta.1",
```
Comment (otorrillas) [inline package.json]: Are we looking for any special feature here that is currently in beta?

File: src/typeform-types.ts [hunk 2]
```diff
@@ -0,0 +1,1233 @@
...
+    ref?: string
+    /**
+     * Array of objects that define the Logic Jump's behavior.
+     */
+    actions?: Action[]
+  }
+  /**
+   * Messages that forms can use.
+   */
+  export interface Messages {
```
Comment (otorrillas) [inline src/typeform-types.ts]: This is amazing, with the descriptions 🎉 

However, is there maybe a way we generate these messages types programatically, if they're prone to change (and I assume it's the API who sets them)?

File: tests/unit/forms.test.ts [hunk 3]
```diff
@@ -85,6 +86,7 @@ test('deleteForm removes the correct uid form ', async () => {
 })
 
 test('create form has the correct path and method ', async () => {
+  // @ts-ignore
```
Comment (otorrillas) [inline tests/unit/forms.test.ts]: Is this line still applicable?

File: package.json (hunk 1 above)
Comment (MichaelSolati) [reply package.json]: Ah, the library would throw an error on node 12. The issue is only resolved on the 4 beta, and not the latest stable release.

File: src/typeform-types.ts (hunk 2 above)
Comment (MichaelSolati) [reply src/typeform-types.ts]: I think the intention is to automate some of this down the road, but time/bandwidth/resources are lacking on my end. I know @picsoung wants to do this. I don't think that the definitions of file structures are likely to change as quickly as the API endpoints are (like new endpoints and new functionality)

File: tests/unit/forms.test.ts (hunk 3 above)
Comment (MichaelSolati) [reply tests/unit/forms.test.ts]: Is it necessary, no... But it prevents a red squiggly line as it's testing an invalid argument... 😅

**Pull request merged**